src/ext/resources/legacygraphqlapi/schema.graphqls

```diff
@@ -758,6 +768,21 @@ enum BikesAllowed {
     NOT_ALLOWED
 }
 
+input BikePreferences {
+    reluctance: Float
+    bikeWalking: WalkPreferences
+    speed: Float
+    switchTime: Duration
+    switchCost: Int
+    optimize: OptimizationCriteria
+    boardCost: Int
+}
```

Comment:
I should also include these switch parameters for scooter preferences but with what name? I checked the code and it appears these are used when switching to bike walking and back to biking. These are seemingly not used when for example when boarding a rented bike for the first time or parking a bike. We have also had a `bikeWalkingReluctance` parameter. Should we combine at least the cost and reluctance into one cost function parameter. Related note, I'm not convinced we should reuse the `WalkPreferences` for bike/scooter walking as it includes board cost which doesnt make sense in these contexts. Also, can we really set different values for bike walking vs regular walking? There are some things for what it could make sense for like the reluctance or the speed as they typically differ a bit.

Reply:
I agree on the WalkPreferences.